src/patternfly/components/Drawer/drawer.scss

```diff
@@ -1,112 +1,244 @@
   }
 
+  // Actions
+  // --pf-c-drawer__actions--PaddingLeft: var(--pf-global--spacer--md);
+  --pf-c-drawer__actions--MarginTop: calc(#{pf-size-prem(6px)} * -1);
+  --pf-c-drawer__actions--MarginRight: calc(#{pf-size-prem(6px)} * -1);
+
   // Box shadow
   --pf-c-drawer__panel--BoxShadow: none;
-  --pf-c-drawer--m-expanded__panel--BoxShadow: var(--pf-global--BoxShadow--md-left);
-  --pf-c-drawer--m-expanded--m-panel-left__panel--BoxShadow: var(--pf-global--BoxShadow--md-right);
+  --pf-c-drawer--m-expanded__panel--BoxShadow: var(--pf-global--BoxShadow--lg-left);
+  --pf-c-drawer--m-expanded--m-panel-left__panel--BoxShadow: var(--pf-global--BoxShadow--lg-right);
 
   // Divider
-  --pf-c-drawer--m-divider__panel--after--Width: var(--pf-global--BorderWidth--sm);
-  --pf-c-drawer--m-divider--m-expanded__panel--after--BackgroundColor: var(--pf-global--BorderColor--100);
+  --pf-c-drawer__panel--after--Width: var(--pf-global--BorderWidth--sm);
+  --pf-c-drawer__panel--after--BackgroundColor: transparent;
+  --pf-c-drawer--m-inline--m-expanded__panel--after--BackgroundColor: var(--pf-global--BorderColor--100);
 
   display: flex;
+  flex-direction: column;
   height: 100%;
   overflow-x: hidden;
+
+  @media screen and (min-width: $pf-global--breakpoint--md) {
+    min-width: var(--pf-c-drawer__panel--MinWidth);
+  }
+}
+
+// Header area
+.pf-c-drawer__section {
+  flex-grow: 0;
+  background-color: var(--pf-c-drawer__section--BackgroundColor);
+
+  &.pf-m-no-background {
+    --pf-c-drawer__section--BackgroundColor: transparent;
+  }
+}
+
+// Main area
+.pf-c-drawer__main {
+  display: flex;
+  flex-grow: 1;
+  overflow: hidden;
 }
 
 // Content / panel
 .pf-c-drawer__content,
 .pf-c-drawer__panel {
+  display: flex;
+  flex-direction: column;
   flex-shrink: 0;
   overflow: auto;
 }
 
 // Content
 .pf-c-drawer__content {
+  --pf-c-drawer--child--PaddingTop: var(--pf-c-drawer__content--child--PaddingTop);
+  --pf-c-drawer--child--PaddingRight: var(--pf-c-drawer__content--child--PaddingRight);
+  --pf-c-drawer--child--PaddingBottom: var(--pf-c-drawer__content--child--PaddingBottom);
+  --pf-c-drawer--child--PaddingLeft: var(--pf-c-drawer__content--child--PaddingLeft);
+
+  z-index: var(--pf-c-drawer__content--ZIndex);
   flex-basis: var(--pf-c-drawer__content--FlexBasis);
   order: 0;
+  background-color: var(--pf-c-drawer__content--BackgroundColor);
+
+  &.pf-m-no-background {
+    --pf-c-drawer__content--BackgroundColor: transparent;
+  }
 }
 
 // Panel
 .pf-c-drawer__panel {
   position: relative;
+  z-index: var(--pf-c-drawer__panel--ZIndex);
   flex-basis: var(--pf-c-drawer__panel--FlexBasis);
   order: 1;
+  overflow: auto;
   background-color: var(--pf-c-drawer__panel--BackgroundColor);
+  box-shadow: var(--pf-c-drawer__panel--BoxShadow);
   transition-duration: var(--pf-c-drawer__panel--TransitionDuration);
   transition-property: var(--pf-c-drawer__panel--TransitionProperty);
   -webkit-overflow-scrolling: touch;
+
+  &::after {
```

Comment:
What is the purpose of this? It doesn't seem to do anything when removed

Reply:
It's the border container. When `.pf-m-border` is applied to parent, the background of this element is painted.